pkg/kubelet/nodestatus/setters_test.go

```diff
@@ -63,16 +64,23 @@ func TestNodeAddress(t *testing.T) {
 		cloudProviderExternal
 		cloudProviderNone
 	)
+	existingNodeAddress := v1.NodeAddress{Address: "10.1.1.2"}
+	hostDefaultIP, err := utilnet.ResolveBindAddress(nil)
```

Comment:
this should not be part of this test , as it make it depend on the host where the test is running

Reply:
You mean the `existingNodeAddress` or `hostDefaultIP` or both ? I know it depends on the host but I thought that when `ResolveBindAddress` will be called under this condition it will be defaulted to call `ChooseHostInterface` and thus whatever is the value returned it should work as  I am re-using this value in the expected result. If you feel it does not make sense, I can try to mock it even if I don't know how yet 😅

Reply:
Will wait to see if CI fails you won't even have to answer 😅

Reply:
Unit test should test the function under test, what is the need to call this external function here?

Reply:
So, we should just assert the behaviour we want on a test, not just focus "on passing ci".

We want that we do X it output y, we can complicate this even more adding external function like this, but it does make sense or is valuable the interface IP address obtained on a worker for tests?

Reply:
I wanted to test the setter behaviour when there is a hostname override without providing a node IP and trigger this code block: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/nodestatus/setters.go#L194-L211
Alright I will remove that part.

Reply:
removed :)